### PR TITLE
move gateway certs specified in server TLS settings to SDS

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/25128

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 1]
```diff
@@ -423,44 +423,22 @@ func buildGatewayListenerTLSContext(
...
-			}
-		}
-		if trustedCa != nil || len(server.Tls.SubjectAltNames) > 0 {
-			ctx.CommonTlsContext.ValidationContextType = &tls.CommonTlsContext_ValidationContext{
-				ValidationContext: &tls.CertificateValidationContext{
-					TrustedCa:            trustedCa,
-					MatchSubjectAltNames: util.StringToExactMatch(server.Tls.SubjectAltNames),
-				},
-			}
+	case server.Tls.Mode == networking.ServerTLSSettings_MUTUAL || server.Tls.Mode == networking.ServerTLSSettings_SIMPLE:
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:433]: Add a default branch

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/gateway.go:433]: It seems this branch is not equal to previous

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/gateway.go]: ok. changed the case to default

**Comment by ramaraochavali** [comment]
>  but it conflicts with what we decided about moving gateway SDS to Istiod.
Sorry. I was not aware of this decision. Closing this

**Comment by costinm** [comment]
BTW - if you have extra time and are interested in SDS - I would appreciate some help with someone taking over https://github.com/istio/istio/pull/25438, moving the gateway SDS in Istiod. I'm a bit too busy right now.

**Comment by ramaraochavali** [comment]
@howardjohn As discussed, rebased and changed. PTAL.

@costinm Discussed with @howardjohn offline. Since DR certs (which are file based), use file based SDS, we need to run SDS at agent event at gateway. Only credential names can use SDS at Istiod. So this PR is still relevant. Please take a look.

**Review by howardjohn — COMMENTED** [review]
Do we have existing integration tests for this?

**Comment by ramaraochavali** [comment]
The existing gateway certs should cover this case and there is an integration test for file mounted certs delivered via SDS.

**Comment by ramaraochavali** [comment]
@costinm Please take a look when you get chance

**Comment by ramaraochavali** [comment]
@costinm Can you PTAL when you get chance and if you are OK, can  you remove the "changes requested"?

**Review by costinm — APPROVED** [review]

**Pull request merged**